libsolidity/inlineasm/AsmCodeGen.cpp

```diff
@@ -217,42 +228,14 @@ class CodeTransform: public boost::static_visitor<>
 		int height = m_state.assembly.deposit();
```

Comment:
I would prefer to keep that option for the programmer and instead perform syntactical checks at a higher level. Basically, there is a subset of "safe assembly" in which jumps and labels cannot be used, pop, swap and dup is forbidden and every statement at block level has to be stack-height neutral (except for VariableDeclaration and similar complex things that already do stack checks internally).